tests/openvino/test_modeling.py

```diff
@@ -1402,3 +1405,113 @@ def test_pipeline(self, model_arch):
         del pipe
         del model
         gc.collect()
+
+
+class OVModelForVision2SeqIntegrationTest(unittest.TestCase):
+    SUPPORTED_ARCHITECTURES = ["vision-encoder-decoder", "trocr", "donut"]
+
+    GENERATION_LENGTH = 100
+    SPEEDUP_CACHE = 1.1
+
+    def _get_sample_image(self):
+        url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+        image = Image.open(requests.get(url, stream=True).raw)
+        return image
+
+    def _get_preprocessors(self, model_id):
+        image_processor = AutoImageProcessor.from_pretrained(model_id)
+        tokenizer = AutoTokenizer.from_pretrained(model_id)
+
+        return image_processor, tokenizer
+
+    def test_load_vanilla_transformers_which_is_not_supported(self):
+        with self.assertRaises(Exception) as context:
+            _ = OVModelForVision2Seq.from_pretrained(MODEL_NAMES["bert"], export=True)
+
+        self.assertIn("only supports the tasks", str(context.exception))
+
+    @parameterized.expand(SUPPORTED_ARCHITECTURES)
+    def test_generate_utils(self, model_arch: str):
+        model_id = MODEL_NAMES[model_arch]
+        model = OVModelForVision2Seq.from_pretrained(model_id, export=True)
+        feature_extractor, tokenizer = self._get_preprocessors(model_id)
+
+        data = self._get_sample_image()
+        features = feature_extractor(data, return_tensors="pt")
+
+        outputs = model.generate(inputs=features["pixel_values"])
+        res = tokenizer.batch_decode(outputs, skip_special_tokens=True)
+        self.assertIsInstance(res[0], str)
+
+        gc.collect()
+
+    @parameterized.expand(SUPPORTED_ARCHITECTURES)
+    def test_model_is_reshapable(self, model_arch: str):
+        model_id = MODEL_NAMES[model_arch]
+        ov_model = OVModelForVision2Seq.from_pretrained(model_id, export=True, compile=False)
+        ov_model.reshape(5, -1)
+        ov_model.compile()
+        models = [ov_model.encoder.model, ov_model.decoder.model, ov_model.decoder_with_past.model]
+        for model in models:
+            for inp_tensor in model.inputs:
+                inp_tensor.partial_shape[0] == 5
```

Comment:
I was thinking about including it in a test like `test_pipeline_image_to_text` to check that it's compatible with inference but we can have the additional test that you added. To not increase too much the time of the github action workflows on push with additional test like `test_model_is_reshapable` I would be in favor of running them nightly instead like https://github.com/huggingface/optimum/blob/dedb852eaea3d3d899920f4017b9fd899eb2756b/tests/exporters/onnx/test_onnx_export.py#L334 and https://github.com/huggingface/optimum/blob/dedb852eaea3d3d899920f4017b9fd899eb2756b/.github/workflows/test_exporters_slow.yml#L33C48-L33C63

Reply:
I do not think that this PR is right place for making such infra changes (because it requires at least make some estimation about which tests should be run on a nightly basis instead of precommit), I also does not think that this specific test makes big impact on testing performance, but if you want I can remove it and add reshape into pipeline tests

Reply:
works for me!